src/gui/owncloudsetupwizard.cpp

```diff
@@ -172,7 +172,10 @@ void OwncloudSetupWizard::slotFoundServer(const QUrl &url, const QJsonObject &in
             auto accountState = AccountState::fromNewAccount(_ocWizard->account()).take(); // we manage the lifetime for this temporary AccountState ourselves (see below)
             if (auto dialog = accountState->updateUrlDialog(url)) {
                 // Dialog is show, make sure the accountState is deleted when the dialog is done.
-                connect(dialog, &QDialog::finished, accountState, &QObject::deleteLater);
+                connect(dialog, &UpdateUrlDialog::finished, accountState, [accountState](int result) {
+                    Q_UNUSED(result);
+                    accountState->deleteLater();
```

Comment:
Do we still need the account state?

Reply:
I don't think so, given it was deleted beforehand, too.

Reply:
We used the account state ptr, to be able to utilise updateUrlDialog.
Now that you refactored it we can use it directly?

Reply:
Indeed, we can just remove the helper entirely and move the code into the new class. That way, we don't need an account state pointer.